packages/sns/lib/utils/snsInitter.ts

```diff
@@ -150,26 +194,30 @@ export async function deleteSns(
     )
   }
 
-  if (!creationConfig.topic.Name) {
-    throw new Error('topic.Name must be set for automatic deletion')
-  }
+  if (creationConfig.topic) {
+    if (!creationConfig.topic.Name) {
+      throw new Error('topic.Name must be set for automatic deletion')
+    }
 
-  await deleteTopic(snsClient, creationConfig.topic.Name)
+    await deleteTopic(snsClient, creationConfig.topic.Name)
```

Comment:
Why this change is required?

Reply:
which part are you referring to?

Reply:
There is extra wrapping condition. I'll simplify it.